tests/test_route53.py

```diff
@@ -101,6 +101,20 @@ def test_route53_hostedzone_markop(self):
         self.assertEqual(len(tags["ResourceTagSet"]["Tags"]), 2)
         self.assertTrue("abc" in tags["ResourceTagSet"]["Tags"][0].values())
 
+    def test_route53_delete_hostedzone(self):
+        session_factory = self.replay_flight_data("test_route53_delete_hostedzone")
+        p = self.load_policy(
+            {
+                "name": "r53domain-delete-hostedzone",
+                "resource": "hostedzone",
+                "filters": [{"tag:TestTag": "present"}],
+                "actions": ["delete"],
+            },
+            session_factory=session_factory,
+        )
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+
```

Comment:
we would generally want to construct a client instance and verify the zone removal by attempting to refetch, if there's a delay on that we generally account for it by doing
```
if self.recording:
    # wait for record deletion
    time.sleep(10)
```

in some cases the resource itself has attribute noting the state transition which we can verify immediately after the call, though that doesn't appear to be the case here.

Reply:
added, thanks!